src/cuda/utils.h

```diff
@@ -21,6 +22,62 @@ void cub_exclusiveSum(IdType* arrays, const IdType array_length) {
                                 arrays, array_length);
 }
 
+template <typename IdType>
+torch::Tensor GetSampledSubIndptr(torch::Tensor indptr, int64_t fanout,
+                                  bool replace) {
+  int64_t size = indptr.numel();
+  auto new_indptr = torch::zeros(size, indptr.options());
+  thrust::device_ptr<IdType> item_prefix(
+      static_cast<IdType*>(new_indptr.data_ptr<IdType>()));
+
+  using it = thrust::counting_iterator<IdType>;
+  thrust::for_each(
+      thrust::device, it(0), it(size),
+      [in_indptr = indptr.data_ptr<IdType>(),
+       out = thrust::raw_pointer_cast(item_prefix), if_replace = replace,
+       num_fanout = fanout] __device__(int i) mutable {
+        IdType begin = in_indptr[i];
+        IdType end = in_indptr[i + 1];
+        if (if_replace) {
+          out[i] = (end - begin) == 0 ? 0 : num_fanout;
+        } else {
+          out[i] = min(end - begin, num_fanout);
+        }
+      });
+
+  cub_exclusiveSum<IdType>(thrust::raw_pointer_cast(item_prefix), size + 1);
+  return new_indptr;
+}
+
+template <typename IdType>
+torch::Tensor GetSampledSubIndptrFused(torch::Tensor indptr,
```

Comment:
I think it would be better to move `GetSampledSubIndptrFused` and `GetSampledSubIndptr` to the new file header `sampling_utils.h`. Because the functions in "utils.h" are used by almost all other operations not only sampling or fused_sampling，but this two functions maybe only used by sampling operations